### PR TITLE
1154 feature add unit test authz

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
#1154

## Summary by Sourcery

Add unit tests for the DataFed Authz library and update the build configuration to support these tests, including new CMake files for GSSAPI and Globus Common libraries.

New Features:
- Introduce unit tests for the DataFed Authz library, including a test for the getVersion function.

Enhancements:
- Add options to the build configuration to enable building of DataFed Authz tests.

Build:
- Include new CMake configuration files for GSSAPI and Globus Common libraries to support building the Authz tests.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,6 +16,7 @@ if( NOT DEFINED GLOBUS_VERSION )
   set(GLOBUS_VERSION  5)
 endif()
 OPTION(BUILD_AUTHZ "Build DataFed Authz library" FALSE)
+OPTION(BUILD_AUTHZ_TESTS "Build DataFed Authz Tests" FALSE)
 OPTION(BUILD_AUTHZ_WITH_SYSLOG "Build Authz without syslog" TRUE)
```
[the remark this answers is not in the document]
Comment (JoshuaSBrown) [reply CMakeLists.txt:20]: I see, how do I leave a default that is false but when the option is used change to true?

[the remark this answers is not in the document]
Comment (JoshuaSBrown) [reply CMakeLists.txt:20]: The example you provided is contradictory to what you said.

**Comment by JoshuaSBrown** [comment]
@par-hermes format

File: cmake/GSSAPI.cmake [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
+function(find_gssapi_library)
+
+  find_library(GSSAPI_LIBRARIES NAMES libgssapi.so.3 libgssapi.so gssapi gssapi )
+
+  if(NOT GSSAPI_LIBRARIES)
```
[the remark this answers is not in the document]
Comment (JoshuaSBrown) [reply cmake/GSSAPI.cmake:6]: Though not a bad suggestion this does not fit with the current context. the Scripts are included and the paths if found are printed without throwing an error. It is only when the GSSAPI library is used that an error needs to be thrown.

File: CMakeLists.txt [hunk 3]
```diff
@@ -125,6 +126,11 @@ if( BUILD_WEB_SERVER )
 
 endif()
 
+if( BUILD_AUTHZ_TESTS )
```
Comment (t-ramz) [inline CMakeLists.txt:129]: I thought these lines were added as part of #1157. Why would these changes show in the diff?

File: cmake/GSSAPI.cmake [hunk 4]
```diff
@@ -0,0 +1,14 @@
+
+function(find_gssapi_library)
+
+  find_library(GSSAPI_LIBRARIES NAMES libgssapi.so.3 libgssapi.so gssapi gssapi )
```
Comment (t-ramz) [inline cmake/GSSAPI.cmake:4]: Looks like this is missing `libgssapi_krb5.so` from #1157

File: repository/gridftp/globus5/authz/CMakeLists.txt [hunk 5]
```diff
@@ -1 +1,5 @@
 add_subdirectory( source )
+
+if( BUILD_AUTHZ_TESTS )
+  add_subdirectory( tests )
```
Comment (t-ramz) [inline repository/gridftp/globus5/authz/CMakeLists.txt:4]: Does `add_subdirectory` tell CMake to search the subdirectory for another List?

**Review by AronPerez — APPROVED** [review]

File: repository/gridftp/globus5/authz/tests/unit/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Each test listed in Alphabetical order
+foreach(PROG
+    test_getVersion
```
Comment (t-ramz) [inline repository/gridftp/globus5/authz/tests/unit/CMakeLists.txt:3]: So is this being used as a string? Is this a variable I'm missing somewhere? I see the filename, but I'm not sure how the `foreach` interprets this list when it's just the unquoted filename.

**Review by t-ramz — COMMENTED** [review]
I added a few questions

File: repository/gridftp/globus5/authz/tests/unit/CMakeLists.txt (hunk 6 above)
Comment (JoshuaSBrown) [reply repository/gridftp/globus5/authz/tests/unit/CMakeLists.txt:3]: It just treats it as a string.

File: repository/gridftp/globus5/authz/CMakeLists.txt (hunk 5 above)
Comment (JoshuaSBrown) [reply repository/gridftp/globus5/authz/CMakeLists.txt:4]: Exactly.

File: cmake/GSSAPI.cmake (hunk 4 above)
Comment (JoshuaSBrown) [reply cmake/GSSAPI.cmake:4]: Good point let me pull in the changes from Dev.

File: CMakeLists.txt (hunk 3 above)
Comment (JoshuaSBrown) [reply CMakeLists.txt:129]: I think that GitHub doesn't always update the file diffs after merging into the target branch. This branch was built off the earlier one let me pull in the changes and see if it shows correctly.

**Review by t-ramz — APPROVED** [review]

**Pull request merged**